### PR TITLE
enhancement: display amount transaction details

### LINKED ISSUE #3900: [Task]: Optimise how we display amount in the transaction details
### Task description

Depending on the amount and decimals we have we might need to optimise how we display the amount in the transaction details page. This may require truncation or rounding, but I would discourage this on the send confirmation form as we want the user to see the exact information they are about to send.

### Requirements

-  Improve how we display amount in the transaction details component

### Acceptance criteria

PR approved and merged

### Creation checklist

- [X] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by jeeanribeiro**
## Summary
Enhances how the AmountBox displays the token amount, by adding smart line breaking, decimals truncation and tooltip to show the full amount with all decimals

## Changelog

```
- Adds line breaking, decimal truncation and tooltip to AmountBox component
```

## Relevant Issues
closes #3900

### Platforms
- __Desktop__
  - [ ] MacOS
  - [x] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions
- Test the send confirmation popup with big amounts and decimals

## Checklist
- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/components/atoms/AmountBox.svelte [hunk 1]
```diff
@@ -1,23 +1,79 @@
...
+        const amountLengthWithoutSeparator = amountWithTwoDecimalNumbers.split(',').join('').length
+        const thousandSeparatorsIndexes: number[] = Array.from(amountWithTwoDecimalNumbers).reduce(
+            (acc, char, i) => (char === getGroupSeparator() ? [...acc, i] : acc),
+            []
+        )
+        const thousandGroups = Math.ceil(amountLengthWithoutSeparator / NUMBERS_PER_THOUSAND_GROUP / LINES)
+
+        const result: string[] = []
+        result.push(amountWithTwoDecimalNumbers.substring(0, thousandSeparatorsIndexes[thousandGroups - 1] + 1))
+        result.push(amountWithTwoDecimalNumbers.substring(result[0].length, amountWithTwoDecimalNumbers.length))
```
Comment (MarkNerdi) [inline packages/shared/components/atoms/AmountBox.svelte:49]: i think you only take care of numbers with a lot of decimal places. If i have an integer which includes a lot of digits, it doesnt break up the string

File: packages/shared/components/atoms/AmountBox.svelte [hunk 2]
```diff
@@ -1,23 +1,79 @@
...
+        <div class="flex flex-col flex-wrap justify-center items-baseline space-x-0.1">
+            {#each displayedAmount as line, index}
+                <div class="flex items-baseline">
+                    <Text type={TextType.h1} fontWeight={FontWeight.semibold}>{line}</Text>
+                    {#if unit && index === displayedAmount.length - 1}
+                        <Text type={TextType.h4} classes="ml-1" fontWeight={FontWeight.medium}>{unit}</Text>
+                    {/if}
+                </div>
+            {/each}
         </div>
```
Comment (MarkNerdi) [inline packages/shared/components/atoms/AmountBox.svelte:72]: cant we rewrite this with some combination of `flex-wrap` and `break-words`?

**Review by MarkNerdi — CHANGES_REQUESTED** [review]
I'm not sure i get what this should do. If i enter a large amount of decimals, they get cropped to 2 places. and if i enter a long integer, it doesnt change anything

File: packages/shared/components/atoms/AmountBox.svelte (hunk 1 above)
Comment (jeeanribeiro) [reply packages/shared/components/atoms/AmountBox.svelte]: Fixed it by adding a new algorithm

File: packages/shared/components/atoms/AmountBox.svelte (hunk 2 above)
Comment (jeeanribeiro) [reply packages/shared/components/atoms/AmountBox.svelte:103]: I tried to do it but I wasn't able to do it, I've also previously tried with `white-space` and it didn't work :(
But if you have a solution I would like to learn from it

Comment (MarkNerdi) [reply packages/shared/components/atoms/AmountBox.svelte:117]: so one solution would be to do it the following way:
```
<Text type={TextType.h1} fontWeight={FontWeight.semibold}>{amount}
   {#if unit }
        <span  class="ml-1">{unit}</span>
   {/if}
</Text>
```

The `span` would need to be styled manually

Comment (MarkNerdi) [reply packages/shared/components/atoms/AmountBox.svelte:117]: another solution would be to give the `h1` containing the amount a `max-widht` + `break-word`

Comment (MarkNerdi) [reply packages/shared/components/atoms/AmountBox.svelte:117]: those solutions have different results, but both valid IMO. So either the unit is always on the right (sol.2) or the unit is part of the unit and added after the last digit, like you implemented (sol.1). Feel free to choose

Comment (jeeanribeiro) [reply packages/shared/components/atoms/AmountBox.svelte:117]: Thank you for your suggestions, I've found a new solution based on yours :bulb: 
```svelte
<Text type={TextType.h1} fontWeight={FontWeight.semibold} classes="whitespace-pre items-baseline">
  {displayedAmount.join('\n')}
  {#if unit}
    <Text type={TextType.h4} classes="inline" fontWeight={FontWeight.medium}>{unit}</Text>
  {/if}
</Text>
```

**Pull request merged**